src/group_theory/perm/sign.lean

```diff
@@ -698,6 +698,73 @@ calc sign f = sign (swap x (f x) * (swap x (f x) * f)) :
       pow_one, units.neg_mul_neg]
 using_well_founded {rel_tac := λ _ _, `[exact ⟨_, measure_wf (λ f, f.support.card)⟩]}
 
+/-- If we have a list of all elements of `α`, we can write
+`equiv.prod_congr_right (e : α → perm β)` as `extend` applied to each element
+in turn. -/
+lemma prod_congr_right_eq_prod_extend (σ : α → perm β)
+  {l : list α} (hl : l.nodup) (mem_l : ∀ a, a ∈ l) :
+  prod_congr_right σ = (l.map (λ a, extend a (σ a))).prod :=
```

Comment:
This lemma statement seems confusing to me - `prod_congr_right` uses `prod` to refer to `×`, but `list.prod` uses it to mean `*`. I can't think of a particularly easy way to avoid this though.

Reply:
Would it be better as `prod_congr_right_eq_list_prod_extend`? Or maybe we can flip the direction of the `eq` sign and call it `prod_(map_)_extend_eq`.

Reply:
I like the idea of flipping the eq